sdk/go/common/workspace/plugins.go

```diff
@@ -1495,26 +1496,14 @@ func cleanupTempDirs(finalDir string) error {
 	return nil
 }
 
-// PluginKind represents a kind of a plugin that may be dynamically loaded and used by Pulumi.
-type PluginKind string
-
-const (
-	// AnalyzerPlugin is a plugin that can be used as a resource analyzer.
-	AnalyzerPlugin PluginKind = "analyzer"
-	// LanguagePlugin is a plugin that can be used as a language host.
-	LanguagePlugin PluginKind = "language"
-	// ResourcePlugin is a plugin that can be used as a resource provider for custom CRUD operations.
-	ResourcePlugin PluginKind = "resource"
-	// ConverterPlugin is a plugin that can be used to convert from other ecosystems to Pulumi.
-	ConverterPlugin PluginKind = "converter"
-	// ToolPlugin is an arbitrary plugin that can be run as a tool.
-	ToolPlugin PluginKind = "tool"
-)
```

Comment:
Do we need to re-export these as well?  Otherwise I think this is still a backwards incompatible change.

Of course that makes it quite awkward to keep in sync,

Reply:
yes I just pushed it. It sucks tho... wonder how can we do it differently 

The source of the circular dependency is this line https://github.com/pulumi/pulumi/pull/15945/files#diff-c59945d33c5fccb80d8b16ea994dfd17192bdd306b70f7feb07c73031fd88f72R855

because apitype depends on workspace in this line https://github.com/pulumi/pulumi/blob/f6b405c24ebc03f293173763ac93a75eacee5d8e/sdk/go/common/apitype/core.go#L356

I could move all the deployment settings definitions under workspace and avoid all this refactor (which I dont like but might simplify everything). What do you think?

Reply:
Yeah, I agree that it kinda sucks.  I guess on second thought we don't actually need to keep them in sync though, if we're introducing new plugin types we can just add it to the `apitype` package, as that won't be a breaking change, and people can move over to that.  

If move the deployment settings definitions under workspace that also gets a bit awkward.

Another potential option would be to create a new package in `pulumi/sdk/go/common/apitype/deployment` and only have the deployment definitions there?  Though I guess that doesn't work either because `DeploymentSettings` would still depend on `apitype`, so we'd just get the same dependency cycle again.  And to break that we'd have to move all of that stuff to apitype/deployment, which means we have to re-export a bunch more stuff, which is not better.

So I guess this is maybe the best we can do unless someone can come up with a smarter idea.

Reply:
@tgummerer so I believe this is ready now. Not sure if the failed `Downstream Codegen Tests` status are something to look into.

Question, should we add a deprecation notice to the re-exported types redirecting people to use apitype?

Reply:
The codegen tests look like they failed to even create the downstream PR, and timed out.  I'm trying to re-run them, but I think we can merge this either way, as it doesn't look like it's actually changing anything in codegen, other than some refactoring, so there's a pretty low chance of something breaking downstream.

I see you did add a deprecation notice already, and I think that's the right place to put it, I don't think we need anything in addition to that.

Reply:
could you merge the pr? I cant (I assume because of the downstream tests)

Reply:
Huh weird, I expected you to be able to merge it either way.  I added it to the merge queue now

Reply:
thank you!

Reply:
Could you merge it again? I just rebased and fixed a new usage of `workspace.ResourcePlugin` that made the lint fail